Description: Store the original ELSA patterndb.xml in etc/elsa/patterns.d/elsa
 .
 securityonion-elsa-extras (20131117-1ubuntu0securityonion2) precise; urgency=low
 .
   * Initial release
Author: Scott Allen Runnels <srunnels@gmail.com>

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: http://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: <YYYY-MM-DD>

--- securityonion-elsa-extras-20131117.orig/bin/securityonion-elsa-config.sh
+++ securityonion-elsa-extras-20131117/bin/securityonion-elsa-config.sh
@@ -112,16 +112,11 @@ function config_lognode() {
     echo "* Adding custom parsers" | $LOGGER
     sh /opt/elsa/contrib/securityonion/contrib/securityonion_parsers_sql.sh
 
-    cp /opt/elsa/node/conf/patterndb.xml /opt/elsa/node/conf/patterndb.xml.bak
-    PRE_MERGE_DIR=/tmp/pre-merge
-    ORIG_PATTERN=/opt/elsa/contrib/securityonion/contrib/securityonion-patterndb.xml.orig
-    NEW_PATTERN_DIR=/etc/elsa/patterns.d
-    MERGED_PATTERN=/opt/elsa/node/conf/patterndb.xml
-    mkdir -p $PRE_MERGE_DIR
-    cp $ORIG_PATTERN $PRE_MERGE_DIR
-    cp -R $NEW_PATTERN_DIR/* $PRE_MERGE_DIR
-    pdbtool merge -p $MERGED_PATTERN --recursive -D $PRE_MERGE_DIR
-    rm -rf $PRE_MERGE_DIR
+    echo "* Merging new parsers into patterndb.xml."
+    PATTERNS_DIR=/etc/elsa/patterns.d/
+    DEST_PATTERN=/opt/elsa/node/conf/patterndb.xml
+    cp $DEST_PATTERN $DEST_PATTERN.bak
+    pdbtool merge -p $DEST_PATTERN --recursive -D $PATTERNS_DIR
 
 	echo "* Placing Sphinx Config file." | $LOGGER
 	# Create the initial sphinx.conf file
